src/main/scala/org/apache/spark/sql/kinesis/KinesisSourceRDD.scala

```diff
@@ -164,12 +170,27 @@ private[kinesis] class KinesisSourceRDD(
             val currentTimestamp: Long = System.currentTimeMillis
             if (canFetchMoreRecords(currentTimestamp)) {
               val shardInterator = getShardIterator()
```

Comment:
we have a flag to control the retries in case of throttling - https://github.com/qubole/kinesis-sql/blob/master/src/main/scala/org/apache/spark/sql/kinesis/KinesisReader.scala#L68

Shall we use that instead of adding the complexity here?

Reply:
This is to ensure that the number of GetRecords call do not exceed more than 5 calls per second similar to the implementation in KCL.

Reply:
@itsvikramagr any update on this please?

Reply:
@PranshuAdhikari - I got stuck with other work. Taking a look at it now.

Reply:
I think we should not add logic to sleep here. We can use  https://github.com/qubole/kinesis-sql/blob/master/src/main/scala/org/apache/spark/sql/kinesis/KinesisReader.scala#L68. 

Logic for retryOrTimeout can be looked again and we can fix it if there are issues.